cmd/zrepl/zrepl.c

```diff
@@ -833,6 +870,56 @@ uzfs_zvol_io_conn_acceptor(void)
 	zk_thread_exit();
 }
 
+/*
+ * This func takes care of sending potentially multiple read blocks each
+ * prefixed by metainfo.
+ */
+static int
+uzfs_send_reads(int fd, zvol_io_cmd_t *zio_cmd)
+{
```

Comment:
indeed. I thought about it and I wanted to put a comment there which I did not do, so I will add it now. We can optimise this in future if it proves to be a bottleneck. I would prefer to use writev so that we don't need to allocate memory and write all data in one syscall, however that requires more changes to uzfs_socket_write/read functions.